### PR TITLE
Add function and chatcommand to get ping to csm and fix RTT calculation.

### LINKED ISSUE #4214: Update RTT in F5 debug info
Strangely enough I've noticed that RTT (round trip time) is updated only on connect in F5 info text, it is useful to check network with it, can you make it update once a second or so as before?

### PR DESCRIPTION
**Pull request by red-001**
Fixes #4214
tested using `tc qdisc add dev lo root handle 1:0 netem delay 150msec`

### CODE REVIEW

File: src/network/connection.cpp [hunk 1]
```diff
@@ -963,11 +967,17 @@ void Peer::RTTStatistics(float rtt, std::string profiler_id,
 		if (jitter >= m_rtt.jitter_max)
 			m_rtt.jitter_max = jitter;
 
-		if (m_rtt.jitter_avg < 0.0)
-			m_rtt.jitter_avg  = jitter;
-		else
-			m_rtt.jitter_avg  = m_rtt.jitter_avg * (num_samples/(num_samples-1)) +
-								jitter * (1/num_samples);
+		m_jitter_list.push_front(jitter);
+		if (m_jitter_list.size() > 15)
```
Comment (SmallJoker) [inline src/network/connection.cpp]: Why 15? `MAX_RELIABLE_WINDOW_SIZE*10` is `327 680` -> way higher

File: src/network/connection.h [hunk 2]
```diff
@@ -732,8 +732,7 @@ class Peer {
 		virtual void reportRTT(float rtt) {};
 
 		void RTTStatistics(float rtt,
-							std::string profiler_id="",
-							unsigned int num_samples=1000);
+							std::string profiler_id = "");
```
Comment (SmallJoker) [inline src/network/connection.h]: Too many indents. Only one (or two) additional ones (to the first function definition line) look good.

File: doc/client_lua_api.md [hunk 3]
```diff
@@ -703,6 +703,8 @@ Call these functions only at load time!
 * `minetest.get_protocol_version()`
     * Returns the protocol version of the server.
     * Might not be accurate at start up as the client might not be connected to the server yet, in that case it will return 0.
+* `minetest.get_ping()`
+    * Returns the round trip time.
```
Comment (SmallJoker) [inline doc/client_lua_api.md]: In seconds?

Comment (red-001) [inline doc/client_lua_api.md]: Yes

File: src/network/connection.cpp [hunk 4]
```diff
@@ -930,22 +930,26 @@ void Peer::DecUseCount()
...
-		else
-			m_rtt.avg_rtt  = m_rtt.avg_rtt * (num_samples/(num_samples-1)) +
-								rtt * (1/num_samples);
+		m_rtt_list.push_front(rtt);
+		if (m_rtt_list.size() > 15)
+			m_rtt_list.pop_back();
+
+		{
+			float total = 0.0;
+			std::list<float>::const_iterator iterator = m_rtt_list.begin();
```
Comment (nerzhul) [inline src/network/connection.cpp]: std::deque cannot be better than a list here ?

File: src/network/connection.h (hunk 2 above)
Comment (Zeno-) [inline src/network/connection.h]: With the other parameters deleted as they are profiler_id will easily fit on line 734

**Comment by paramat** [comment]
@red-001 needs work, still interested?

**Comment by paramat** [comment]
Maybe the useful RTT bugfix part can be split off and adopted.

**Comment by paramat** [comment]
```
+core.register_chatcommand("ping", {
+	description = core.gettext("Get ping"),
+	func = function(param)
+		local ping = math.floor(core.get_ping() * 1000 + 0.5)
+		core.display_chat_message(core.gettext("Ping: ") .. ping .. "ms")
+	end,
+})
```
Is ping actually useful? In another PR we are considering it useless, or is this a different ping?

**Comment by rubenwardy** [comment]
Please may you separate the RTT calculation fix from the CSM feature?

**Comment by paramat** [comment]
Keeping open, i think the bugfix is worth splitting off and adopting.

**Comment by paramat** [comment]
Yes, please split, the RTT fix is welcome.
Neglected so closing for now.

**Pull request closed without merging**